app/src/main/java/ch/eureka/eurekapp/model/connection/ConnectivityObserver.kt

```diff
@@ -35,6 +35,27 @@ open class ConnectivityObserver internal constructor(context: Context) {
                   override fun onUnavailable() {
                     trySend(false)
                   }
+
+                  override fun onLost(network: Network) {
+                    // Network is completely lost, check current state
+                    val currentNetwork = connectivityManager.activeNetwork
+                    val capabilities = connectivityManager.getNetworkCapabilities(currentNetwork)
+                    val isCurrentlyConnected =
+                        capabilities?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ==
+                            true
+                    trySend(isCurrentlyConnected)
```

Comment:
why not make this one block i.e.
```suggestion        
                    trySend(
                       connectivityManager
                           .activeNetwork
                           ?.hasCapability(
                               NetworkCapabilities.NET_CAPABILITY_INTERNET
                           ) ?: false
                    )
```
or change `isCurrentlyConnected` to this or extract it to a function as both `onCapabilitiesChanged` and `onLost` reuse some code.
```suggestion
                    val isCurrentlyConnected =
                        capabilities?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ?: false
```